lib/ansible/modules/windows/setup.ps1

```diff
@@ -88,142 +90,135 @@ for ($i=1; $i -le ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProc
-Set-Attr $date "tz_offset" $datetime.ToString("zzzz")
-Set-Attr $date "tz" ([System.TimeZoneInfo]::Local.Id)
-# Ensure that the weekday is in English
-Set-Attr $date "weekday" $datetime.ToString("dddd", [System.Globalization.CultureInfo]::InvariantCulture)
-Set-Attr $date "weekday_number" (Get-Date -UFormat "%w")
-Set-Attr $date "weeknumber" (Get-Date -UFormat "%W")
-Set-Attr $date "year" $datetime.ToString("yyyy")
-Set-Attr $result.ansible_facts "ansible_date_time" $date
-
-# Win32_PhysicalMemory is empty on some virtual platforms
-Set-Attr $result.ansible_facts "ansible_memtotal_mb" ([math]::round($win32_cs.TotalPhysicalMemory / 1024 / 1024))
-Set-Attr $result.ansible_facts "ansible_swaptotal_mb" ([math]::round($win32_os.TotalSwapSpaceSize / 1024 / 1024))
-
-Set-Attr $result.ansible_facts "ansible_lastboot" $win32_os.lastbootuptime.ToString("u")
-Set-Attr $result.ansible_facts "ansible_uptime_seconds" $([System.Convert]::ToInt64($(Get-Date).Subtract($win32_os.lastbootuptime).TotalSeconds))
+
+$date = ${
+    date = $datetime.ToString("yyyy-MM-dd")
+    day = $datetime.ToString("dd")
+    epoch = (Get-Date -UFormat "%s")
+    hour = $datetime.ToString("HH")
+    iso8601 = $datetime_utc.ToString("yyyy-MM-ddTHH:mm:ssZ")
+    iso8601_basic = $datetime.ToString("yyyyMMddTHHmmssffffff")
+    iso8601_basic_short = $datetime.ToString("yyyyMMddTHHmmss")
+    iso8601_micro = $datetime_utc.ToString("yyyy-MM-ddTHH:mm:ss.ffffffZ")
+    minute = $datetime.ToString("mm")
+    month = $datetime.ToString("MM")
+    second = $datetime.ToString("ss")
+    time = $datetime.ToString("HH:mm:ss")
+    tz = ([System.TimeZoneInfo]::Local.Id)
+    tz_offset = $datetime.ToString("zzzz")
+    # Ensure that the weekday is in English
+    weekday = $datetime.ToString("dddd", [System.Globalization.CultureInfo]::InvariantCulture)
+    weekday_number = (Get-Date -UFormat "%w")
+    weeknumber = (Get-Date -UFormat "%W")
+    year = $datetime.ToString("yyyy")
+}
 
 $ips = @()
-Foreach ($ip in $netcfg.IPAddress) { If ($ip) { $ips += $ip } }
-Set-Attr $result.ansible_facts "ansible_ip_addresses" $ips
+Foreach ($ip in $netcfg.IPAddress) {
+    If ($ip) {
+        $ips += $ip
+    }
+}
 
-$env_vars = New-Object psobject
-foreach ($item in Get-ChildItem Env:)
-{
+$env_vars = @{ }
+foreach ($item in Get-ChildItem Env:) {
     $name = $item | select -ExpandProperty Name
     # Powershell ConvertTo-Json fails if string ends with \
     $value = ($item | select -ExpandProperty Value).TrimEnd("\")
-    Set-Attr $env_vars $name $value
+    $env_vars.Add($name, $value)
 }
-Set-Attr $result.ansible_facts "ansible_env" $env_vars
 
-$psversion = $PSVersionTable.PSVersion.Major
-Set-Attr $result.ansible_facts "ansible_powershell_version" $psversion
+$ansible_facts = @{
+    ansible_architecture = $win32_os.OSArchitecture
+    ansible_bios_date = $win32_bios.ReleaseDate.ToString("MM/dd/yyyy")
+    ansible_bios_version = $win32_bios.SMBIOSBIOSVersion
+    ansible_date_time $date
```

Comment:
From the CI output, it looks like you need
`ansible_date_time = $date `
on L 137